File: python/src/eventc.py
#! /usr/bin/env python

import argparse

from gi.repository import GLib
from gi.repository import Eventc
from gi.repository import EventdEvent as Eventd

parser = argparse.ArgumentParser(description='Basic Python CLI client for eventd')

parser.add_argument('-H', '--host', default='localhost', help='Host to connect to')
parser.add_argument('-w', '--wait', action='store_true', help='Wait the end of the event')
parser.add_argument('-d', '--data', nargs=2, action='append', help='Event data (name, content) to send')
parser.add_argument('category', help='Event category to send')
parser.add_argument('name', help='Event name to send')
args = parser.parse_args()

eventc = Eventc.Connection.new(args.host)
eventc.set_passive(not args.wait)

event = Eventd.Event.new(args.category, args.name)

for d in args.data:
    event.add_data(d[0], d[1])

loop = GLib.MainLoop.new(None, False)

def disconnect(data1 = None, data2 = None):
    print "disconnect"
    eventc.close()
    loop.quit()

def connect(data):
    if args.wait:
        event.connect('ended', disconnect)

    eventc.connect_sync()
    eventc.event(event)

    if not args.wait:
        disconnect(None)

    return False


GLib.idle_add(connect, None)

loop.run()

File: lua/src/eventc.lua
#! /usr/bin/env lua

local lgi = require 'lgi'
local GLib = lgi.GLib
local Eventd = lgi.EventdEvent
local Eventc = lgi.Eventc

local host = 'localhost'
local wait = false
local data = {}
local category = nil
local name = nil

while ( #arg > 0 ) do
    local a = arg[1]
    table.remove(arg, 1)
    if ( ( a == '-h' ) or ( a == '--host' ) ) then
        host = arg[1]
        table.remove(arg, 1)
    elseif ( ( a == '-w' ) or ( a == '--wait' ) ) then
        wait = true
    elseif ( ( a == '-d' ) or ( a == '--data' ) ) then
        data[arg[1]] = arg[2]
        table.remove(arg, 1)
        table.remove(arg, 1)
    elseif ( category == nil ) then
        category = a
    elseif ( name == nil ) then
        name = a
    else
        usage()
    end
end

local eventc = Eventc.Connection.new(host)
eventc:set_passive(not wait)

local event = Eventd.Event.new(category, name)

if ( #data > 0 ) then
    for k, v in data do
        event:add_data(k, v)
    end
end

local loop = GLib.MainLoop.new(nil, false)

function disconnect(data1, data2)
    print("disconnect")
    eventc:close()
    loop:quit()
end

function connect(data)
    if wait then
        event:connect('ended', disconnect)
    end

    eventc:connect_sync()
    eventc:event(event)

    if not wait then
        disconnect(nil)
    end

    return false
end

GLib.idle_add(GLib.PRIORITY_DEFAULT_IDLE, connect, nil)

loop:run()
